packages/lesswrong/components/form-components/FormComponentPostEditorTagging.tsx

```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+import { Components, registerComponent } from '../../lib/vulcan-lib';
+import toDictionary from '../../lib/utils/toDictionary';
+import mapValues from 'lodash/mapValues';
+import { taggingNamePluralCapitalSetting } from '../../lib/instanceSettings';
+
+/**
+ * Edit tags on the new or edit post form. If it's the new form, use
+ * TagMultiSelect; a server-side callback will convert tags to tag-relevances.
+ * If it's the edit form, instead use FooterTagList, which has the same
+ * voting-on-tag-relevance as the post page. Styling doesn't match between these
+ * two, which is moderately unfortunate.
+ */
+const FormComponentPostEditorTagging = ({value, path, document, label, placeholder, formType, updateCurrentValues}: {
+  value: any,
+  path: string,
+  document: any,
+  label?: string,
+  placeholder?: string,
+  formType: "edit"|"new",
+  updateCurrentValues: any,
+}) => {
+  const { CoreTagsChecklist, TagMultiselect, FooterTagList } = Components
+  if (formType === "edit") {
+    return <FooterTagList
+      post={document}
+      hideScore
+      hidePostTypeTag
+      showCoreTags
+    />
+  } else {
+    return <div>
+      <CoreTagsChecklist existingTagIds={Object.keys(value||{})} onTagSelected={(tag)=> {
+        updateCurrentValues(
+          mapValues(
+            {tagRelevance:[tag.tagId]},
+            (arrayOfTagIds: string[]) => toDictionary(
+              arrayOfTagIds, tagId=>tagId, tagId=>1
+            )
```

Comment:
I feel like this should be turned into a function with a comment explaining why we're doing this

Reply:
I have pinged Jim.

Reply:
I think the gist is that the post `tagRelevance` field needs to look like `{string: number}`, so even though it's extra work both here and in the callback, we need to maintain that structure.